httpx/dispatch/http2.py

```diff
@@ -125,6 +133,9 @@ def initiate_connection(self) -> None:
             self.timeout_flags[stream_id].set_read_timeouts()
             if isinstance(event, h2.events.ResponseReceived):
                 break
+            elif isinstance(event, h2.events.WindowUpdated):
+                if event.stream_id and event.stream_id in self.flow_control_events:
+                    self.flow_control_events[event.stream_id].set()
```

Comment:
So what if the server gives us a WindowUpdated event but the size updated isn't large enough for us to be able to send? I think we need to track the threshold we're waiting for in addition to the `BaseEvent` object, decrement the threshold we're waiting for each time we receive an event, and then only call `.set()` when we're past the threshold.

Am I understanding that properly?

Reply:
So. Okay. I understand the problem here, but I'm not sure that we should solve it this way. I think it would be better to use new window size that server gave us, and then wait for new updates of the window. So for example done in `hypercorn`

Reply:
Also about the waiting for the required value. The [RFC 7540](https://tools.ietf.org/html/rfc7540#section-5.2.1) in the flow control section says that the client can request any window size, but it is the receiver who decides what to provide to the client:

> A receiver MAY choose to set any window size that it
       desires for each stream and for the entire connection.  A sender
       MUST respect flow-control limits imposed by a receiver.

Based on this, I think that we just need to use what is given to us, and then again wait for the WindowUpdated event.

Reply:
Right, but isn't the current implementation going to wait for any window size update and then send our pre-determined chunk size that we have? If we're going off of what our peer is telling us to use as a window shouldn't we have to recalculate our chunk size whenever we are ready to send the next chunk?

The code as I'm seeing it right now is doing the following:
1. Calculate chunk size once
2. Create an iterator of chunks from the above chunk size. (Not recalculated)
3. If we can't send the current chunk we have then we wait for any WindowUpdated event
4. Even if that WindowUpdated event didn't increase our available window to be large enough for that chunk we send anyways?

Reply:
Yes, you are right. I changed the implementation of the `.send_data` method, so can you check current version, please.